src/widgets/activation/index.tsx

```diff
@@ -28,20 +30,30 @@ export const Activation = () => {
     const handleSuccess = () => {
       setMessages((messages) => [
         {
-          message: 'Почта подтверждена',
+          message: 'Ваш Email успешно подтвержден',
           type: ApiMessageTypes.success,
         },
         ...messages,
       ]);
     };
-    if (user && user?.email !== '') {
+    if (user?.email !== '' && !user?.is_active) {
       api
         .activateEmail(uid || '', token || '')
         .then(handleSuccess)
         .catch(handleError)
-        .finally(() => navigate('/', { replace: true }));
+        .finally(() => {
+          navigate('/', { replace: true });
+          setIsLoading && setIsLoading(false);
+        });
+    } else if (user?.is_active) {
+      handleSuccess();
+      setIsLoading && setIsLoading(false);
+    } else {
+      setIsLoading && setIsLoading(false);
     }
-  }, [user, uid, token, navigate, setMessages]);
+    //NOTE: Disabling deps check to use useEffect only on mount
```

Comment:
то, что передать пустой массив сделает useEffect вызываться только on mount это понятно. Это написано в реактовых доках и у нас все в реакте шарят. То есть это как бы в этой строчке уже написано это, поэтому нет смысла писать это в комменте.

А вот что не написано, это почему вы хотите запустить этот эффект только on mount. И почему не подходит передать сюда setIsLoading

```suggestion
    // Load user data on mount only. setIsLoading is instantiated twice [написать почему]
```